### ISSUE TITLE
How does open layers calculate resolutions for each zoom?

### ISSUE THREAD

**Issue by taigi100** [openlayers/ol3#4482]
Considering I have the following code : 

    var mapExtent = [0.000000, -14721.000000, 11896.000000, 0.000000];
    var mapMinZoom = 0;
    var mapMaxZoom = 5;
    var mapMaxResolution = 1.000000;
    var tileExtent = [0.000000, -14721.000000, 11896.000000, 0.000000];

    var mapResolutions = [];
    for (var z = mapMinZoom; z <= mapMaxZoom; z++) {
      mapResolutions.push(Math.pow(2, mapMaxZoom - z) * mapMaxResolution);  
    }

How does OL3 calculate the full size for each zoom level ?

**Comment by ahocevar** [comment]
For view coordinates, It uses the extent(s) you specify for the tile grid. If you want to know the size in pixel, just divide by the resolution. Resolution is map units per pixel.

What exactly do you want to achieve?

**Comment by taigi100** [comment]
I'm trying to display a huge image using OL3. At first, I used maptiler to break them down and it went amazing, but now that I got to testing with even bigger images I can't use it anymore. I want to be able to use some custom resolutions for example : 
Zoom 0 : 3.1%
Zoom 1 : 6.25%
Zoom 2 : 12.5%
Zoom 3 : 25%
Zoom 4: 50%
Zoom 5: 75%
Zoom 6: 100%
Zoom 7: 125%
Zoom 8: 150%
Zoom 9: 175%
Zoom 10: 200%
Zoom 11: 300%

Or so. If I would define it this way it would be easy to calculate each resolution, and breaking it down should be pretty straight forward too. Sadly I can't seem to figure out how to define those resolutions inside.
